src/background.c

```diff
@@ -262,6 +262,122 @@ handle_notify_background (XdpImplBackground *object,
   return TRUE;
 }
 
+/* This is conservative, but lets us avoid escaping most
+   regular Exec= lines, which is nice as that can sometimes
+   cause problems for apps launching desktop files. */
+static gboolean
+need_quotes (const char *str)
+{
+  const char *p;
+
+  for (p = str; *p; p++)
+    {
+      if (!g_ascii_isalnum (*p) &&
+          strchr ("-_%.=:/@", *p) == NULL)
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+static char *
+maybe_quote (const char *str)
+{
+  if (need_quotes (str))
+    return g_shell_quote (str);
+  return g_strdup (str);
+}
+
+static gboolean
+handle_enable_autostart (XdpImplBackground *object,
+                         GDBusMethodInvocation *invocation,
+                         const char *arg_app_id,
+                         gboolean    arg_enable,
+                         const char * const *arg_commandline,
+                         gboolean    arg_activatable)
+{
+  gboolean result = FALSE;
+  g_autofree char *dir = NULL;
+  g_autofree char *file = NULL;
+  g_autofree char *path = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GString) args = NULL;
+  g_autoptr(GKeyFile) keyfile = NULL;
+
+  g_debug ("background: handle EnableAutostart");
+
+  file = g_strconcat (arg_app_id, ".desktop", NULL);
```

Comment:
Do we ever want callers to be able to generate multiple autostart files? I.e. do we want to pass in some kind of suffix for the filename?

Reply:
I don't see a need